src/replication.c

```diff
@@ -3586,6 +3614,14 @@ static int rdbChannelSendHandshake(connection *conn, sds *err) {
             serverLog(LL_WARNING, "Error sending AUTH to master in rdb channel replication handshake: %s", *err);
             return C_ERR;
         }
+    } else if (server.cluster_enabled) {
```

Comment:
For RDB channel, there is no issue when there are mixed versions in cluster, since both RDB channel and internal secret are introduced from 8.0.

If the replica goes here, the master supports the rdb channel, which means it supports the internal secret too.